pywbem_mock/_mainprovider.py

```diff
@@ -0,0 +1,4209 @@
+        if not include_class_origin:
+            self._remove_classorigin(rtn_inst)
+        return rtn_inst
+
+    def _get_subclass_list_for_enums(self, classname, namespace, class_store):
+        """
+        Get class list (i.e names of subclasses for classname for the
+        enumerateinstance methods in a case-insenstive dictionary. The input
+        classname is included in this dictionary.
+
+        Returns:
+           NocaseDict where only the keys are important, This allows case
+           insensitive matches of the names with Python "for cln in clns".
+
+        Raises:
+
+            CIMError: (CIM_ERR_INVALID_CLASS) if classname not in CIM repository
+        """
+
+        if not class_store.exists(classname):
+            raise CIMError(
+                CIM_ERR_INVALID_CLASS,
+                _format("Class {0!A} not found in namespace {1!A}.",
+                        classname, namespace))
+
+        clnslist = self._get_subclass_names(classname, class_store, True)
+        clnsdict = NocaseDict()
+        for cln in clnslist:
+            clnsdict[cln] = cln
+
+        clnsdict[classname] = classname
+        return clnsdict
+
+    @staticmethod
+    def _filter_properties(obj, property_list):
+        """
+        Remove properties from an instance or class that aren't in the
+        property_list parameter
+
+        obj(:class:`~pywbem.CIMClass` or :class:`~pywbem.CIMInstance):
+            The class or instance from which properties are to be filtered
+
+        property_list(list of :term:`string`):
+            List of properties which are to be included in the result. If
+            None, remove nothing.  If empty list, remove everything. else
+            remove properties that are not in property_list. Duplicated names
+            are allowed in the list and ignored.
+        """
+        if property_list is not None:
+            property_list = [p.lower() for p in property_list]
+            for pname in obj.properties.keys():
+                if pname.lower() not in property_list:
+                    del obj.properties[pname]
+
+    ###############################################################
+    #
+    #   Access  to data store
+    #
+    #   The following methods provide access to the data store for
+    #   a single namespace. There are separate methods to provide
+    #   access to each of the 3 object stores within each namespace of
+    #   the CIM Repository CIMClass, CIMInstance, and CIMQualifierDeclaration.
+    #
+    ##################################################################
+
+    def get_class_store(self, namespace):
+        """
+        Returns the class object store for the specified CIM namespace within
+        the CIM repository.  This method validates that the namespace exists in
+        the data store.
+
+        This method does not validate that the namespace exists. Be certain
+        namespace is validated against CIM repository before calling this
+        method
```

Comment:
The method it calls is documented to raise `KeyError` if the namespace does not exist, even in the base class that defines the interface.

I suggest to shorten the (meanwhile conflicting) statements about namespace validation to "The namespace must exist in the CIM repository". The "Raises" section details what happens if that condition is not met.

Reply:
Changed to use your sentence, statementt that namespace should be validated before calling this method. Finally added KeyError as raises.

DONE